### ISSUE TITLE
High memory usage issue

### ISSUE THREAD

**Issue by alexisbar** [mixpanel/mixpanel-swift#563]
Hello,

I faced a memory issue in my app that is related to MixPanel's swift SDK.
This has been really visible when I started using an extension, as the memory limit is really low in these lands. (24Mb for a Notification Service Extension)
After some debugging I found that it was due to a call that was made upon each process launch:

`Mixpanel.mainInstance().people.unset(properties: ["myAwesomeProperty"])`

When receiving a lot of notification in a row, this was causing my extension to hit the memory limit.

This is easily reproductible in any application by implementing these lines of code:

```
for _ in 1...10000 {
    Mixpanel.mainInstance().people.unset(properties: ["myAwesomeProperty"])
}
```

When triggered upon app launch, this causes a really high memory usage

![Capture d’écran 2022-10-17 à 14 51 05](https://user-images.githubusercontent.com/8861304/196181556-dc8c4600-432a-4161-8b79-6550092ac572.png)

iOS 16
Mixpanel-swift (4.0.2) installed via cocoa pods

**Comment by adamwulf** [comment]
I'm seeing the same issue. When offline and tracking many events, the memory grows significantly.

**Comment by zihejia** [comment]
hi @adamwulf , your fix has been merged and is in 4.1.0. Thanks so much again! I'm closing this one now. Feel free to reopen it anytime.
